scripts/preprocessor.py

```diff
@@ -32,7 +31,29 @@ def preprocess(
         filters: 'dict', maps attributes to filter functions that we want to apply,
         decimal_accuracy: 'int', how many decimal digits to consider for floats
     '''
-    data = pd.read_csv(data_file_name)
+    xml = ET.parse(data_file_name)
+    root = xml.getroot()
+    
+    string = "\{(.*?)\}"
+    prefix = re.search(string,root.tag)[0]
+
+    columns = {}
+
+    for tname in root.findall('.//' + prefix + 'ClinicalVariables'):
+
+        attribute = tname.find(prefix + 'TypeName').text
+        if attribute not in columns.keys():
+            columns[attribute] = []
+        try:
```

Comment:
Create a function called `convertToType(value)` that converts and returns the type.